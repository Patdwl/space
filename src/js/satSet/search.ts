import { SatObject } from '../api/keepTrackTypes';
import { SpaceObjectType } from '../api/SpaceObjectType';

export const year = (satData: SatObject[], yr: number) =>
  satData.filter((sat) => {
    const tleYear = sat?.TLE1?.substring(9, 11) || '-1';
    return parseInt(tleYear) == yr;
  });
export const yearOrLess = (satData: SatObject[], yr: number) =>
  satData.filter((sat) => {
    const tleYear = sat?.TLE1?.substring(9, 11) || '-1';
    if (yr >= 59 && yr < 100) {
      return parseInt(tleYear) <= yr && parseInt(tleYear) >= 59;
    } else {
      return parseInt(tleYear) <= yr || parseInt(tleYear) >= 59;
    }
  });
export const name = (satData: SatObject[], regex: RegExp): SatObject[] => satData.filter((sat) => !!regex.test(sat.name));
export const country = (satData: SatObject[], regex: RegExp): SatObject[] => satData.filter((sat) => !!regex.test(sat.country));
export const shape = (satData: SatObject[], text: string): SatObject[] => satData.filter((sat) => sat.shape === text);
export const bus = (satData: SatObject[], text: string): SatObject[] => satData.filter((sat) => sat.bus === text);
export const type = (satData: SatObject[], objType: SpaceObjectType): SatObject[] => satData.filter((sat) => sat.type === objType);

export const search = {
  year,
  yearOrLess,
  name,
  country,
  shape,
  bus,
  type,
};
