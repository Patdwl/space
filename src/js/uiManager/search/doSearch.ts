import { keepTrackApi } from '@app/js/api/keepTrackApi';
import { legendMenuChange } from '..';
import { searchBox } from './searchBox';
import { toast } from '../ui/toast';
import { updateURL } from '../url/update-url';

export const doSearch = (searchString: string, isPreventDropDown = false) => {
  if (searchString == '') {
    searchBox.hideResults();
    return;
  }

  const { satSet } = keepTrackApi.programs;

  let idList = searchBox.doSearch(searchString, isPreventDropDown);
  if (idList.length === 0) {
    if (settingsManager.lastSearch?.length > settingsManager.minimumSearchCharacters) {
      toast('No Results Found', 'serious', false);
    }
    searchBox.hideResults();
    return;
  }

  if (settingsManager.isSatOverflyModeOn) {
    satSet.satCruncher.postMessage({
      typ: 'satelliteSelected',
      satelliteSelected: idList,
    });
  }
  // Don't let the search overlap with the legend
  legendMenuChange('clear');
  updateURL();
};
