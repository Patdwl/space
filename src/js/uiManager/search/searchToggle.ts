import { getEl } from '@app/js/lib/helpers';
import { hideSideMenus } from '../../plugins/breakup/breakup';
import { searchBox } from './searchBox';

let isSearchOpen = false;
let forceClose = false;
let forceOpen = false;
export const searchToggle = (force?: boolean) => {
  // Reset Force Options
  forceClose = false;
  forceOpen = false;

  // Pass false to force close and true to force open
  if (typeof force != 'undefined') {
    if (!force) forceClose = true;
    if (force) forceOpen = true;
  }

  if ((!isSearchOpen && !forceClose) || forceOpen) {
    isSearchOpen = true;
    getEl('search-holder').classList.remove('search-slide-up');
    getEl('search-holder').classList.add('search-slide-down');
    getEl('search-icon').classList.add('search-icon-search-on');
    getEl('fullscreen-icon').classList.add('top-menu-icons-search-on');
    getEl('tutorial-icon').classList.add('top-menu-icons-search-on');
    getEl('legend-icon').classList.add('top-menu-icons-search-on');
  } else {
    isSearchOpen = false;
    getEl('search-holder').classList.remove('search-slide-down');
    getEl('search-holder').classList.add('search-slide-up');
    getEl('search-icon').classList.remove('search-icon-search-on');
    setTimeout(function () {
      getEl('fullscreen-icon').classList.remove('top-menu-icons-search-on');
      getEl('tutorial-icon').classList.remove('top-menu-icons-search-on');
      getEl('legend-icon').classList.remove('top-menu-icons-search-on');
    }, 500);
    hideSideMenus();
    searchBox.hideResults();
    // getEl('menu-space-stations').classList.remove('bmenu-item-selected');
    // This is getting called too much. Not sure what it was meant to prevent?
    // satSet.setColorScheme(colorSchemeManager.default, true);
    // uiManager.colorSchemeChangeAlert(settingsManager.currentColorScheme);
  }
};
