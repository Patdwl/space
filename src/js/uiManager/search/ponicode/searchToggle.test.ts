import * as searchToggle from '@app/js/uiManager/search/searchToggle';
// @ponicode
describe('searchToggle.searchToggle', () => {
  test('0', () => {
    let result: any = searchToggle.searchToggle(true);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = searchToggle.searchToggle(false);
    expect(result).toMatchSnapshot();
  });
});
