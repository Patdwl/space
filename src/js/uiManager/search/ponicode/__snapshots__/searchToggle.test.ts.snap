// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchToggle.searchToggle 0 1`] = `undefined`;

exports[`searchToggle.searchToggle 1 1`] = `undefined`;
